scalyr_agent/config_main.py

```diff
@@ -1132,6 +1134,53 @@ def create_custom_dockerfile(
     out_tar.close()
 
 
+PYTHON2 = "python2"
+PYTHON3 = "python3"
+
+
+def set_python_version(version):
+    """Switch agent command main files to another version of python"""
+    controller = PlatformController.new_platform()
+
+    if controller.install_type != PACKAGE_INSTALL:
+        raise Exception("Not a package.")
+
+    binary_path = os.path.join("/", "usr", "share", "scalyr-agent-2", "bin")
+    source_path = os.path.join(
+        "/", "usr", "share", "scalyr-agent-2", "py", "scalyr_agent"
+    )
+
+    if version == PYTHON3:
+        agent_main_filename = "agent_main_py3.py"
+        config_main_filename = "config_main_py3.py"
+    else:
+        agent_main_filename = "agent_main_py2.py"
+        config_main_filename = "config_main_py2.py"
+
+    agent_main_source = os.path.join(source_path, agent_main_filename)
+    config_main_source = os.path.join(source_path, config_main_filename)
+
+    scalyr_agent_2_target = os.path.join(binary_path, "scalyr-agent-2")
+    scalyr_agent_2_config_target = os.path.join(binary_path, "scalyr-agent-2-config")
+
+    def make_symlink(source, target):
+        try:
+            os.symlink(source, target)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                os.remove(target)
+                os.symlink(source, target)
+
+    make_symlink(agent_main_source, scalyr_agent_2_target)
+    make_symlink(config_main_source, scalyr_agent_2_config_target)
+
+    print("Switched agent to python {0}".format(version))
+    print(
+        "If you have an existing instance of scalyr-agent-2 process running, "
+        "you need to restart it for this change to take an affect."
```

Comment:
We can also add, "You can do that by running ``/etc/init.d/scalyr-agent-2 restart``" or similar.